feat: implement condition 2 #4

Storsorken opened the pull request:
Check that NUMPOINTS is 3 or greater is needed so that we don't get index out of bounds.

Closes #4

Storsorken committed:
feat: implement condition 2 #4

Check that NUMPOINTS is 3 or greater is needed so that we don't get index out of bounds.

paulinegarelli commented on decide/src/main/java/org/example/Decide.java:
```diff
@@ -55,7 +55,9 @@ public boolean condition1() {
-        return true;
+        return settings.NUMPOINTS >= 3 && IntStream.range(0, settings.NUMPOINTS - 2).anyMatch(
+                (index) -> (settings.POINTS[index + 1].angle(settings.POINTS[index], settings.POINTS[index + 2])
+                        > Math.PI - settings.PARAMETERS.EPSILON));
```
I think we're missing the other case here, it's either:
angle > PI + EPSILON
or
angle < PI - EPSILON

Storsorken replied:
Thanks for the catch!

paulinegarelli reviewed (changes requested):
Please change the comparison of the angle to include both cases, otherwise looks good!

paulinegarelli committed:
Merge branch 'main' into erik/feat/4-add-condition2

Storsorken committed:
Merge branch 'main' into erik/feat/4-add-condition2

Storsorken committed:
bug: fix condition2 #52 @paulinegarelli

Condition 2 did not previously check both conditions for the angles nor check that points did not coincide with middle point, see #4

hampfh reviewed (changes requested):
sorry approved by accident

hampfh reviewed (approved):
👍
